src/main/java/individual/freshplace/service/MemberService.java

```diff
@@ -0,0 +1,50 @@
+package individual.freshplace.service;
+
+import individual.freshplace.dto.SignupRequest;
+import individual.freshplace.entity.DiscountByGrade;
+import individual.freshplace.entity.Member;
+import individual.freshplace.repository.DiscountByGradeRepository;
+import individual.freshplace.repository.MemberRepository;
+import individual.freshplace.util.ErrorCode;
+import individual.freshplace.util.constant.GradeCode;
+import individual.freshplace.util.exception.DuplicationException;
+import individual.freshplace.util.exception.WrongValueException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class MemberService {
+
+    private final MemberRepository memberRepository;
+    private final DiscountByGradeRepository discountByGradeRepository;
+    private final BCryptPasswordEncoder bCryptPasswordEncoder;
```

Comment:
```suggestion
    private final PasswordEncoder PasswordEncoder;
```
구현체에 의존하지 않아도된다고 생각하는데 어떻게 생각하세요?
그리고 제가 말씀드리는 구현체에 의존하지 않아야된다라는 이야기는 왜 나온것이며 무슨 장점이 있을까요?

Reply:
> 구현체에 의존하지 않아도된다고 생각하는데 어떻게 생각하세요? 그리고 제가 말씀드리는 구현체에 의존하지 않아야된다라는 이야기는 왜 나온것이며 무슨 장점이 있을까요?

구현체에 의존할 경우 구체화에 의존하지 않고 추상화에 의존해야 한다는 DIP원칙에 위반됩니다. 스프링에서는 DI컨테이너를 통해  의존관계를 주입할 수 있으므로 구현체를 변경하고싶다면 외부에서 구현체를 주입하고 내부에서는 추상화를 의존하여 내부코드의 변경없이 확장이 가능한 객체지향설계원칙을 지킬 수 있다는 장점이 있습니다. 해당 코드는 변경하도록 하겠습니다.